### ISSUE TITLE
Allow violation with any severity to be used in tests

### ISSUE THREAD

**Issue by ThatSneakyCoder** [checkstyle/checkstyle#12980]
Identified at https://github.com/checkstyle/checkstyle/pull/12884#issuecomment-1505458409

>ERROR] [checkstyle] [ERROR] /home/vsts/work/1/s/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariableTestWarningSeverity.java:0: All check input files in the 'src/(it|test)/resources*/' folder must                    have trailing comments "// ok" and/or "// violation". [StyleValidationCommentInInputs]

### Current scenario
For an input file like this:
```java
/*
UnusedLocalVariable
severity = warning


*/

package com.puppycrawl.tools.checkstyle.checks.coding.unusedlocalvariable;

public class InputUnusedLocalVariableTestWarningSeverity {

    void m() {
     @Test.A Outer p1 = new @Test.A Outer();
     @Test.A Outer.@Test.B Inner p2 = p1.new @Test.B Inner();
    }

}
```
where ``severity = warning``, we get ``StyleValidationCommentInInputs`` error which is depicted below:
```
[INFO] [checkstyle] Running Checkstyle  on 3231 files
[INFO]      [echo] Checkstyle finished (checkstyle-resources-checks.xml): 13/04/2023 10:36:10 PM
[INFO]      [echo] Checkstyle started (checkstyle-input-checks.xml): 13/04/2023 10:34:41 PM
[INFO] [checkstyle] Running Checkstyle  on 2057 files
[ERROR] [checkstyle] [ERROR] C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\coding\unusedlocalvariable\InputUnusedLocalVariableTestWarningSeverity.java:0: All check input files in the 'src/(it|test)/resources*/' folder must                    have trailing comments "// ok" and/or "// violation". [StyleValidationCommentInInputs]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  08:41 min
[INFO] Finished at: 2023-04-13T22:37:24+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (ant-phase-verify) on project checkstyle: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\config\ant-phase-verify.xml:190: Got 1 errors and 0 warnings.
[ERROR] around Ant part ...<ant antfile="config/ant-phase-verify.xml" />... @ 6:50 in C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\target\antrun\build-main.xml
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

With the current BDD model, we are not able to write ``// violation ...`` in input file if ``severity`` is set to ``warning``. 

### What is expected 

We should be able to write ``// violation`` in input files irrespective of what the ``severity`` is set to. This requires re-work of the BDD model. Below is how the input file should look like with the violations on the respective lines:

```java
/*
UnusedLocalVariable
severity = warning


*/

package com.puppycrawl.tools.checkstyle.checks.coding.unusedlocalvariable;

public class InputUnusedLocalVariableTestWarningSeverity {

    void m() {
     @Test.A Outer p1 = new @Test.A Outer();
     @Test.A Outer.@Test.B Inner p2 = p1.new @Test.B Inner(); // violation ..
    }

}
```

**Comment by nrmancuso** [comment]
@shubh220922 please show what we expect as result; that even when we use different severity, we should consider it a violation. Remove all mention of CI, these links expire and CI has nothing to do with this issue. We need to update test methods/BDD to support different severity levels, please investigate and add details to issue description about this.

**Comment by romani** [comment]
We should not demand "ok" as it is noise.
We should not demand `violation` also as we already almost done with migration to use this model of input for all.

We need to remove this validation check in our config.

**Comment by romani** [comment]
fixed by https://github.com/checkstyle/checkstyle/pull/17392
